store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/integration/FirstLastIT.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.accumulostore.integration;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.accumulo.core.client.admin.CompactionConfig;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import uk.gov.gchq.gaffer.accumulostore.AccumuloProperties;
+import uk.gov.gchq.gaffer.accumulostore.AccumuloStore;
+import uk.gov.gchq.gaffer.commonutil.StreamUtil;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.graph.Graph;
+import uk.gov.gchq.gaffer.integration.StandaloneIT;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.data.EntitySeed;
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.io.Output;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.SchemaEntityDefinition;
+import uk.gov.gchq.gaffer.store.schema.TypeDefinition;
+import uk.gov.gchq.koryphe.impl.binaryoperator.First;
+import uk.gov.gchq.koryphe.impl.binaryoperator.Last;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.gchq.gaffer.commonutil.TestGroups.ENTITY;
+import static uk.gov.gchq.gaffer.commonutil.TestPropertyNames.PROP_1;
+import static uk.gov.gchq.gaffer.commonutil.TestPropertyNames.PROP_2;
+import static uk.gov.gchq.gaffer.store.TestTypes.ID_STRING;
+import static uk.gov.gchq.gaffer.store.TestTypes.PROP_INTEGER;
+import static uk.gov.gchq.gaffer.store.TestTypes.PROP_INTEGER_2;
+import static uk.gov.gchq.gaffer.store.TestTypes.STRING_TYPE;
+
+class FirstLastIT extends StandaloneIT {
```

Comment:
Thinking about this again, perhaps this should go into core Gaffer ITs rather than a standalone AccumuloIT

Reply:
Due to how this interacts specifically with Accumulo compactions, I don't see any reason this should affect any other store.